frontend/www/js/omegaup/components/contest/Intro.vue

```diff
@@ -0,0 +1,282 @@
+            </form>
+          </div>
+        </template>
+
+        <template v-else="">
+          <!-- Must login to do anything -->
+          <div class="card">
+            <div class="card-body">
+              <p>{{ T.mustLoginToJoinContest }}</p>
+              <a v-bind:href="redirectURL" class="btn btn-primary">{{
+                T.loginHeader
+              }}</a>
+            </div>
+          </div>
+        </template>
+      </div>
+      <hr />
+      <div>
+        <h1>{{ T.registerForContestChallenges }}</h1>
+        <p>{{ contest.description }}</p>
+      </div>
+      <div>
+        <h1>{{ T.registerForContestRules }}</h1>
+        <ul>
+          <li v-if="contest.show_scoreboard_after">
+            {{ T.contestNewFormScoreboardAtContestEnd }}
+          </li>
+          <li v-if="contest.window_length !== null">
+            {{ differentStartsDescription }}
+          </li>
+          <li>{{ scoreboardDescription }}</li>
+          <li>{{ submissionsGapDescription }}</li>
+          <li>{{ penaltyTypes[contest.penalty_type] }}</li>
+          <li v-if="contest.penalty !== 0">{{ penaltyDescription }}</li>
+          <li>{{ feedbackTypes[contest.feedback] }}</li>
+          <li v-if="contest.points_decay_factor !== 0">
+            {{ pointsDecayDescription }}
+          </li>
+        </ul>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import { omegaup } from '../../omegaup';
+import T from '../../lang';
+import * as markdown from '../../markdown';
+import * as UI from '../../ui';
+import omegaup_Countdown from '../Countdown.vue';
+
+interface Statement {
+  gitObjectId?: string;
+  markdown?: string;
+  statementType?: string;
+}
+
+interface ContestStatement {
+  contest_alias: string;
+  share_user_information: boolean | null;
+  privacy_git_object_id?: string;
+  statement_type?: string;
+}
+
+@Component({
+  components: {
+    'omegaup-countdown': omegaup_Countdown,
+  },
+})
+export default class ContestIntro extends Vue {
+  @Prop() contest!: omegaup.Contest;
+  @Prop() isLoggedIn!: boolean;
+  @Prop() requestURI!: string;
+  @Prop() requestsUserInformation!: string;
+  @Prop() needsBasicInformation!: boolean;
+  @Prop() statement!: Statement;
+
+  T = T;
+  UI = UI;
+  penaltyTypes = {
+    none: T.contestNewFormNoPenalty,
+    problem_open: T.contestNewFormByProblem,
+    contest_start: T.contestNewFormByContests,
+    runtime: T.contestNewFormByRuntime,
+  };
+  feedbackTypes = {
+    detailed: T.contestNewFormImmediateFeedbackDesc,
+    none: '',
+    summary: T.contestNewFormImmediateSummaryFeedbackDesc,
+  };
+  markdownConverter = markdown.markdownConverter();
+  shareUserInformation = null;
+  now = Date.now();
+
+  get redirectURL(): string {
+    const url = encodeURIComponent(window.location.pathname);
+    return `/login/?redirect=${url}`;
+  }
+  get consentHtml(): string {
+    if (!this.statement) {
```

Comment:
```suggestion
    if (!this.statement?.markdown) {
```

así L195 no necesita el `|| ''`. Alternativamente

```suggestion
    const markdown = this.statement?.markdown;
    if (!markdown) {
        return '';
    }
```

Reply:
Listo, utilicé la alternativa